Jenkinsfile

```diff
@@ -11,14 +11,36 @@ max_time = 120
 // assign any caught errors here
 err = null
 
+//Method to Kill PR Builds that are running currently when a new build is triggered.
+def abortPreviousRunningBuilds() {
+  def hi = Hudson.instance
+  def pname = env.JOB_NAME.split('/')[0]
+
+  hi.getItem(pname).getItem(env.JOB_BASE_NAME).getBuilds().each{ build ->
+    def exec = build.getExecutor()
+
+    if (build.number != currentBuild.number && exec != null) {
```

Comment:
I have two questions here:

1. Does this apply to merge builds as well? 

2. What happens let's say you have 3 builds for the same PR, build #1, #2 and #3 and let's say the order in which the executor is acquired is #1, #3, #2. When #2 runs after #3. This code will kill #3 and start #2, it should only kill jobs whose build numbers are lower?

Reply:
1. I believe it does. Merge builds run on master, correct? I should definitely add a condition on the function call saying if 'branch_name is not master'. I wanted to test what happens on apache before adding that

2. Do the PR builds not get executors in FCFS order? In that case wouldnt build #2 always get an executo rbefore build #3? (However, I can think of scenarios where the sanity test of build #3 runs befiore build #2 and adding this condition seems like a good idea)